crates/ty_python_semantic/resources/mdtest/annotations/self.md

```diff
@@ -301,6 +301,28 @@ reveal_type(Container("a"))  # revealed: Container[str]
 reveal_type(Container(b"a"))  # revealed: Container[bytes]
 ```
 
+## Implicit self for classes with a default value for their generic parameter
+
+```py
+from typing import Self, TypeVar, Generic
+
+class Container[T = bytes]:
+    def method(self) -> Self:
+        return self
+
+def _(c: Container[str]):
+    reveal_type(c.method())  # revealed: Container[str]
```

Comment:
Would it be helpful to add tests that fall back on the default specialization?

Reply:
It's not really related to this bug at all, but maybe it makes the test a bit more comprehensive for general regression-catching? Sure, I can add that.